test/fhirtypes/ElementDefinition.constrainType.test.ts

```diff
@@ -182,6 +242,113 @@ describe('ElementDefinition', () => {
+      const expectedType = new ElementDefinitionType('Identifier').withProfiles(
+        'http://example.org/identifiers/StructureDefinition/required-type-identifier',
+        'http://example.org/identifiers/StructureDefinition/required-use-identifier',
+        'http://example.org/identifiers/StructureDefinition/required-system-value-identifier'
+      );
+      expectedType._profile = [
+        {
+          id: 'type-extra',
+          extension: [
+            {
+              url: 'http://hl7.org/fhir/StructureDefinition/elementdefinition-type-must-support',
+              valueBoolean: true
+            }
+          ]
+        },
+        null,
+        null
+      ];
+      expect(identifier.type).toHaveLength(1);
+      expect(identifier.type[0]).toEqual(expectedType);
+    });
+
+    it('should null multiple _profile entries when a profile is constrained to multiple more specific profiles', () => {
+      const identifier = modifiedObservation.elements.find(e => e.id === 'Observation.identifier');
+      const identifierConstraint = new OnlyRule('entry.resource');
+      identifierConstraint.types = [
+        { type: 'required-type-identifier' },
+        { type: 'required-use-identifier' },
+        { type: 'required-system-value-identifier' },
+        { type: 'required-type-value-identifier' }
+      ];
+      identifier.constrainType(identifierConstraint, fisher);
+      const expectedType = new ElementDefinitionType('Identifier').withProfiles(
+        'http://example.org/identifiers/StructureDefinition/required-type-identifier',
+        'http://example.org/identifiers/StructureDefinition/required-use-identifier',
+        'http://example.org/identifiers/StructureDefinition/required-system-value-identifier',
+        'http://example.org/identifiers/StructureDefinition/required-type-value-identifier'
+      );
+      expectedType._profile = [
+        {
+          id: 'type-extra',
+          extension: [
+            {
+              url: 'http://hl7.org/fhir/StructureDefinition/elementdefinition-type-must-support',
+              valueBoolean: true
+            }
+          ]
+        },
+        null,
+        null,
+        null
+      ];
+      expect(identifier.type).toHaveLength(1);
+      expect(identifier.type[0]).toEqual(expectedType);
+    });
+
+    it('should remove _profile aray when only null values are left', () => {
+      const identifier = modifiedObservation.elements.find(e => e.id === 'Observation.identifier');
+      const identifierConstraint = new OnlyRule('entry.resource');
+      identifierConstraint.types = [{ type: 'required-use-identifier' }];
+      identifier.constrainType(identifierConstraint, fisher);
+      expect(identifier.type).toHaveLength(1);
+      expect(identifier.type[0]).toEqual(
+        new ElementDefinitionType('Identifier').withProfiles(
+          'http://example.org/identifiers/StructureDefinition/required-use-identifier'
+        )
+        // NOTE: no _profile array
```

Comment:
The assertion would not really do anything for us -- because we're already asserting `equals` on `identifier.type[0]` -- so if `identifier.type[0]._profile` existed, that `equals` assertion would fail.

I just put that comment to call out that I was constructing the expected type _without_ the `_profile`.

So... I think it's OK as it is.